libraries/chain/protocol/restriction.cpp

```diff
@@ -217,8 +217,12 @@ struct attr_argument_validation_visitor
          // validate common data
          r.validate_common_data();
          // validate member-related data
-         restriction_validation_visitor vtor( r );
-         vtor( *((const T*)nullptr) );
+
+//TODO: uncomment when tests for functionality will be ready
+//      commmented to fix build on macos
```

Comment:
There are currently a few fixes worked on for compile issues on MacOSX, please let us know you issues and we might be able to link you to the issue numbers

Reply:
Actually I plan to remove this code after review (whole file).
This logic was mostly moved to new files, this are Abit's implementation, I left it for review purposes.